### PR TITLE
Add cohort review overview page

### PR DESCRIPTION
**Pull request by tjennison-google**
* Generalize action bar controls so that different ones can be displayed on different pages.
* Generalize text input using a form so keyboard controls (and eventually validation) work correctly.
* Add routing and links to/from cohort page.
* Add basic operations.

TODO

* Backend requests. This will require require a significant extension to the integration test infrastructure and I had already started this PR before the backend APIs were available.
* Stats.
* Form validation.

### CODE REVIEW

File: ui/src/router.tsx [hunk 1]
```diff
@@ -42,12 +47,18 @@ export function AppRouter() {
...
 
+// TODO(tjennison): This is becoming spaghetti. Consider alternative ways to set
+// this up or perhaps alternative libraries.
 export function underlayURL(underlayName: string) {
   return underlayName;
 }
 
-export function cohortURL(cohortId: string, groupId: string) {
-  return "cohorts/" + cohortId + "/" + groupId;
+export function cohortURL(cohortId: string, groupId?: string) {
```
Comment (marikomedlock) [inline ui/src/router.tsx:56]: Is this function called from somewhere other than `absoluteCohortURL` below? If not, maybe you can inline it there instead.

File: ui/src/cohortReview/cohortReviewList.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    initialText: selectedReview?.displayName,
+    textLabel: "Review name",
+    buttonLabel: "Rename",
+    onConfirm: onRenameReview,
+  });
+
+  return (
+    <>
+      <ActionBar
+        title={`Reviews of ${cohort.name}`}
```
Comment (marikomedlock) [inline ui/src/cohortReview/cohortReviewList.tsx:114]: Nit: Cohort reviews or Reviews of Cohort {name}?

File: ui/src/data/source.tsx [hunk 3]
```diff
@@ -970,3 +1035,86 @@ function normalizeRequestedAttributes(attributes: string[]) {
     ),
   ];
 }
+
+function fromAPICohort(cohort: tanagra.CohortV2): tanagra.Cohort {
```
Comment (marikomedlock) [inline ui/src/data/source.tsx:1039]: Do you plan to keep this translation between the API and UI schemas long-term, or is it just a short-term bridge to keeping the existing code working? On the backend, I separate the API serialization from the internal objects, just wondering if you do the same thing for UI code.

File: ui/src/cohortToolbar.tsx [hunk 4]
```diff
@@ -1,19 +1,38 @@
...
+} from "hooks";
 import { Link as RouterLink } from "react-router-dom";
 import { ActionCreators as UndoActionCreators } from "redux-undo";
+import { cohortReviewURL } from "router";
+
+export default function CohortToolbar() {
+  const underlay = useUnderlay();
+  const cohort = useCohort();
 
-function UndoRedo() {
```
Comment (marikomedlock) [inline ui/src/cohortToolbar.tsx:9]: Why did you remove the `UndoRedo` toolbar function? I see that it's replaced with `CohortToolbar` in the other files as well. Is it just always included everywhere the `Cohorts` toolbar button is so you bundled them?

**Review by marikomedlock — APPROVED** [review]

File: ui/src/cohortReview/cohortReviewList.tsx (hunk 2 above)
Comment (tjennison-google) [reply ui/src/cohortReview/cohortReviewList.tsx:114]: I think that's unnecessarily verbose given the context of how you get to this page and the fact that the names are user selected.

File: ui/src/cohortToolbar.tsx (hunk 4 above)
Comment (tjennison-google) [reply ui/src/cohortToolbar.tsx:9]: Yeah, I need to specify the same set of buttons for the `ActionBar` everywhere within the cohort editor so it's easiest to group them within a component. I could have made a new one that also included the `UndoRedo` component but it's also only used within the cohort editor and I don't expect it to be used elsewhere so it made more sense to just merge them together.
